elisp-bytecode.texi

```diff
@@ -334,7 +334,15 @@ Discard one value.
 @node byte-dup
 @unnumberedsubsec @code{byte-dup} (137)
 @kindex byte-dup
-Duplicate one value.
+Push a copy the top of the evaluation stack
+
+@subsubsection Example
+When lexical binding is in effect, @code{(defun en(n) n)} generates:
+@verbatim
+PC  Byte  Instruction
+ 0  137   (byte-dup . 0)  ;; duplicates top of stack: n
```

Comment:
I think it's great this works, but we should put a footnote somewhere to let readers know optimization is disabled for our examples.

Reply:
That reminds me, that I've not been able to get the custom instructions for constant values from 0..63 to appear, but instead we git the 192 opcode version with the 16-bit value. I wonder if that's due to lack of optimization.

Reply:
Sorry if I'm misunderstanding the issue, but note that 0..63 is the index into the constant vector. If you have just one constant in the function it's always going to be opcode 192 (which doesn't take a 16-bit value). Or are you seeing opcode 129 plus a 16-bit index which is less than 64?

Reply:
```@subsubsection Example
@code{(defun six() 6)} generates:
@verbatim
PC  Byte  Instruction
 0  192   (byte-constant 6)
 1  135   (byte-return . 0)
@end verbatim

@unnumberedsubsec @code{byte-constant} (192--255)
@kindex byte-constant
Pushes a constant reference on the stack.
Note that there are special instructions to push constants from 0 to 63.

@subsubsection Example
@code{(defun mm() 2000)} generates:
@verbatim
PC  Byte  Instruction
 0  192   (byte-constant 2000)
 1  135   (byte-return . 0)
@end verbatim
```

Why is the opcode value of the byte-constant the same value 192?

Reply:
192 means "take the first constant in the constant vector and push it on the stack". If you have just one constant, it'll always be opcode 192.

To see opcode 193, use two different constants; for 129, you'll have to use at least 64 constants, I'm afraid.

Reply:
Many thanks. I'll use 3 or so constants and then folks will get the idea.  I don't think there is value in showing all 64 opcodes in use.